lektor/types/formats.py

```diff
@@ -17,3 +23,45 @@ class MarkdownType(Type):
 
     def value_from_raw(self, raw):
         return MarkdownDescriptor(raw.value or u"")
+
+
+# Wrapper with an __html__ method prevents Lektor from escaping HTML tags.
+class HTML:
+    def __init__(self, html):
+        self.html = html
+
+    def __html__(self):
+        return self.html
+
+
+class HTMLDescriptor:
+    def __init__(self, source):
+        self.source = source
+
+    def __get__(self, obj, type=None):
```

Comment:
This adds a lot of new magic to `HtmlType`.
Aside from concerns about rewriting links to pages to force them to the PRIMARY_ALT, this will probably result in surprises to existing users, who will find their links suddenly mangled.

I'm not convinced this is the way to go, at all.  But if it is, this functionality should probably be enabled by a new _model option_ for the `html` type so as not to break existing sites.